packages/flutter/lib/src/widgets/route_name.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+/// A widget which provides a semantic name route name.
+///
+/// See also:
+///
+///   * [SemanticsProperties.route], for a description of how route name
+///     semantics work.
+class RouteName extends StatelessWidget {
+
+  /// Creates a widget which provides a semantic route name.
+  ///
+  /// [child] and [name] are required arguments.
+  const RouteName({
+    Key key,
+    @required this.child,
+    @required this.name,
+  }) : assert(child != null),
+       assert(name != null),
+       super(key: key);
+
+  /// The widget below this widget in the tree.
+  ///
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  /// A semantic name for the route.
+  final String name;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Semantics(
+      route: true,
+      explicitChildNodes: true,
+      value: name,
+      child: child,
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(new StringProperty('name', name, defaultValue: ''));
```

Comment:
There's no default currently, so we should probably not specify one here.

But maybe we should have a default value?

The way I'd love to use this widget is just to wrap it with a single "child" argument, which automatically grabs the semantic value of the child and uses that as the default name. (Except maybe on iOS it should default to no value? I don't understand what iOS does exactly.)

Reply:
Spoke offline with @yjbanov  and @goderbauer  on this - it doesn't seem like that would give very good results as a default. For example, the child semantics are merged and contain multiple headers, or ordered in such a way that we can't determine what the right value to use is. 

And since we introducing a new Widget here and not inserting it into the navigator or pages, it makes sense to just require it.